README.md

```diff
@@ -147,27 +145,12 @@ spec:
   should be excluded in [objectives][16].
 - **timeWindow[ ]** *TimeWindow* is a list but accepting only exactly one
   item, one of the rolling or calendar aligned time window:
-
   - Rolling time window. Minimum duration for rolling time window is 5
-    minutes, maximum 31 days).
-
-    ```yaml
-    unit: Day | Hour | Minute
-    count: numeric
-    isRolling: true
-    ```
-
+    minutes, maximum 31 days). Duration should be provided in shorthand format
+    e.g. 5m, 31d.
   - Calendar Aligned time window. Minimum duration for calendar aligned time
-  window is 1 day and maximum is 366 days.
-
-  ```yaml
-  unit: Year | Quarter | Month | Week | Day
-  count: numeric
-  calendar:
-      startTime: 2020-01-21 12:30:00 # date with time in 24h format
-      timeZone: America/New_York # name as in IANA Time Zone Database
-  # isRolling: false # for calendar aligned set false value or not set
-  ```
+    window is 1 day and maximum is 366 days. Duration should be provided in shorthand format
+    eg. 1d, 366d.
```

Comment:
Ah this one might be tricky.  Not sure if the standard duration shorthand libraries respect a `Quarterly` designation, but we'll need something to represent that.  
Maybe 3 _months_ is sufficient? Quarters are going to be different in total number of Days or Weeks, so I don't think it's safe to use those.
Is it weird of Calendar Aligned durations continue to use a more verbose object?  The Timezone information is important to keep too.

Reply:
Yeah, the duration for windows is tricky. We have two types of windows
- rolling
- calendar aligned

For rolling windows duration can be expressed as we have already in spec as `Day | Hour | Minute`. Those units always mean the same and we can think about expressing them as `1d | 1h | 1m` maybe can consider introducing an additional one - `Week - 1w` (because it always equals `7d`) or just expect to use `7d`. By allowing only those units, we don't have any unambiguity. In contrast to `1 Month` which length differs e.g. between February and March. Of course, we can introduce some artificial normalized `Month` e.g. which always equals `30d` and `Year` equals `365d`, but it introduces additional cognitive load for a user. Thus maybe it's better just to force her to use `30d` or `365d` directly in a spec. 

For calendar aligned everything become more complicated
```yaml
unit: Year | Quarter | Month | Week | Day
count: numeric # count of time units for example count: 7 and unit: Day means 7 days window
calendar:
  startTime: 2020-01-21 12:30:00 # date with time in 24h format, format without time zone
  timeZone: America/New_York # name as in IANA Time Zone Database
```
First of all, it doesn't make sense to have units smaller than `1 Day`. Moreover, the length of each unit depends on the timezone and time of the year. Thus we require `startTime` to know how to make calculations. When we start 1st February one month is counted to the last day of February, the next window starts from the 1st March to the end of March, etc. It is aligned with the calendar and makes it easy to use for instance SLO as a monthly SLA for a service. The exact math may be [differently implemented by libraries/languages](https://dev.to/nordcloud/how-a-calculation-of-previous-month-in-golang-can-break-production-1ppi). For instance Nobl9 decided to use the Golang way. The same is for `Quarter` (which will include months with different numbers of days) and `Year` (which can be a leap year). Furthermore by considering `timeZone` in math daylight saving is included too. Thus even a particular `Day` may vary in the number of hours in it. But by considering everything from the above, an SLO window will match watches and calendars used by businesses (people). The open question is do we want and how to define a short form for a unit in this scenario without making conflicts with units defined for the rolling window.